components/common/src/liveliness_checker.rs

```diff
@@ -165,22 +167,38 @@ fn unregister_thread_impl<T, E: ToString>(_: CheckedThread,
 /// Call once per binary to start the thread which will check that all the threads that
 /// call `mark_thread_alive` continue to do so.
 pub fn spawn_thread_alive_checker() {
-    thread::Builder::new().name("thread-alive-check".to_string())
-                          .spawn(|| -> ! {
-                              let delay = ThreadAliveCheckDelay::configured_value().into();
+    spawn_thread_alive_checker_impl(ThreadAliveCheckDelay::configured_value().into());
+}
+
+fn spawn_thread_alive_checker_impl(delay: Duration) {
+    let executable = std::env::args_os().next()
+                                        .map(PathBuf::from)
+                                        .and_then(|first_arg| {
+                                            first_arg.components().last().map(|basename| {
+                                                                             basename.as_os_str()
+                                                                                 .to_string_lossy()
+                                                                                 .into_owned()
+                                                                         })
+                                        })
+                                        .unwrap_or_else(|| "unknown-executable".into());
+
+    debug!("Spawning liveliness checker thread for {} PID {}",
+           executable,
+           std::process::id());
+    let thread_name = format!("{}-liveliness-checker", executable);
+    thread::Builder::new().name(thread_name.clone())
+                          .spawn(move || -> ! {
+                              debug!("Starting {}", thread_name);
+
                               let threshold = ThreadAliveThreshold::configured_value().into();
                               let max_time_since_death =
                                   ThreadDeadIgnoreDelay::configured_value().into();
                               loop {
-                                  {
-                                      let statuses =
-                                          &mut THREAD_STATUSES.lock()
-                                                              .expect("THREAD_STATUSES poisoned");
-                                      check_thread_heartbeats(statuses, threshold);
-                                      log_dead_threads(statuses);
-                                      cull_dead_threads(statuses, max_time_since_death);
-                                  }
                                   thread::sleep(delay);
+                                  let statuses = &mut THREAD_STATUSES.lock();
```

Comment:
heh... sneaky order swap here to avoid the explicit scoping :smile:

Reply:
The benefit of adding tests is it justifies more sneakiness 😉

Reply:
Sneakiness-Driven Development!